add serializers pseudo

p-galligan opened the pull request:
Adds pseudo code for tsv/csv serializers.

Fixes #23

Commit:
add serializers pseudo

Fixes #23 . Adds pseudo code for tsv/csv serializers.

Commit:
linting changes

Updates for linting compliance.

helrond reviewed (commented):
Can we get the linters to pass on this?

Otherwise, I think this is fine, but we may want to discuss a couple of things:
- I think this is something that could be done in a much more object-oriented ways, i.e. a Serializer class which has some methods on it, and is maybe subclassed by a CSVSerializer and a TSVSerializer
- I recently learned about the [DictWriter](https://docs.python.org/3/library/csv.html#csv.DictWriter) class in Python's CSV module, which seems like a promising option

Commit:
syntax update

Fixing syntax

Commit:
updating to pass linting

linting updates

Commit:
linting update

updates syntax for linting

Commit:
Update to pass linting

I know this isn't OOP, but I've wracked my brain about how to do that, and I'm a little confused right now.
